### PR TITLE
Feature/APMI-3297-Getting an identifier and carrier name directly ios

### PR DESCRIPTION
**Pull request by Piyush23786**
CoreTeleohony framwork added

### CODE REVIEW

File: SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift [hunk 1]
```diff
@@ -67,6 +67,12 @@ func endHttpSpan(span: Span?, task: URLSessionTask) {
     if hostConnectionType != nil {
         span?.setAttribute(key: "net.host.connection.type", value: hostConnectionType!)
     }
+    if hostConnectionSubtype != nil {
+        span?.setAttribute(key: "net.host.connection.subtype", value: hostConnectionSubtype!)
+    }
+    if hostConnectionName != nil {
+        span?.setAttribute(key: "net.host.carrier.name", value: hostConnectionName!)
+    }
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift:75]: Since the nil check is done on the first line, perhaps use `span!` instead of `span?`

File: SplunkRumWorkspace/SplunkRum/SplunkRum.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -612,6 +612,7 @@
 				SKIP_INSTALL = YES;
 				SUPPORTS_MACCATALYST = YES;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_PRECOMPILE_BRIDGING_HEADER = NO;
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum.xcodeproj/project.pbxproj:615]: Is this addition necessary? Same for the other `SWIFT_PRECOMPILE_BRIDGING_HEADER` line

File: SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkType.swift [hunk 3]
```diff
@@ -31,13 +35,84 @@ func initializeNetworkTypeMonitoring() {
...
     }
+}
+
+ func getNetworkInfo() -> String? {
 
+     if #available(iOS 12.0, *) {
+         guard let dict = networkInfo.serviceCurrentRadioAccessTechnology else {
+             return nil
+         }
+         if dict.count != 0 {
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkType.swift:76]: Is it possible here are multiple networks in use at the same time here? 🤔

Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkType.swift:76]: Dictionary containing the current radio access technology registered to each service.
1.Physical SIM card.
2.eSIM.

File: SplunkRumWorkspace/SplunkRum/SplunkRum.xcodeproj/project.pbxproj (hunk 2 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum.xcodeproj/project.pbxproj:615]: sorry its not necessary, i will revert this.

File: SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift (hunk 1 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift]: Done.
